master/pom.xml

```diff
@@ -1789,6 +1789,16 @@
 
     <plugins>
       <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-eclipse-plugin</artifactId>
+        <version>2.10</version>
+        <!-- https://issues.apache.org/jira/browse/MECLIPSE-726 -->
+        <configuration>
+          <useProjectReferences>true</useProjectReferences>
+          <workspace>..</workspace>
```

Comment:
I'm not 100% sure this will fix is as when you're running mvn eclipse:eclipse in a subfolder it will point to a different location to when you're running it at the root of the checkout.